addon/lib/metrics.js

```diff
@@ -65,90 +59,71 @@ const variantMaker = {
 const Metrics = module.exports = {
 
   init: function() {
-    if (!store.telemetryPingPayload) {
-      store.lastTelemetryPingTimestamp = false;
-      store.telemetryPingPayload = {
-        version: 1,
-        tests: {}
-      };
-    }
-
-    Metrics.maybePingTelemetry();
-
+    store.browserLoadedTimestamp = Date.now();
     Events.on(EVENT_SEND_METRIC, Metrics.onExperimentPing);
     Events.on(EVENT_RECEIVE_VARIANT_DEFS, Metrics.onReceiveVariantDefs);
   },
 
   onEnable: function() {
-    // Backup existing preference settings and then override.
-    store.metricsPrefsBackup = {};
-    Object.keys(PREFERENCE_OVERRIDES).forEach(name => {
-      store.metricsPrefsBackup[name] = PrefsService.get(name);
-      PrefsService.set(name, PREFERENCE_OVERRIDES[name]);
-    });
+    Metrics.pingTelemetry(self.id, 'enabled', Date.now());
+    Metrics.prefs.backup();
   },
 
   onDisable: function() {
-    // Restore previous preference settings before override.
-    if (store.metricsPrefsBackup) {
+    Metrics.pingTelemetry(self.id, 'disabled', Date.now());
+    Metrics.prefs.restore();
+  },
+
+  prefs: {
+    // Backup existing preference settings and then override.
+    backup: function() {
+      store.metricsPrefsBackup = {};
       Object.keys(PREFERENCE_OVERRIDES).forEach(name => {
-        PrefsService.set(name, store.metricsPrefsBackup[name]);
+        store.metricsPrefsBackup[name] = PrefsService.get(name);
+        PrefsService.set(name, PREFERENCE_OVERRIDES[name]);
       });
+    },
+
+    // Restore previous preference settings before override.
+    restore: function() {
+      if (store.metricsPrefsBackup) {
+        Object.keys(PREFERENCE_OVERRIDES).forEach(name => {
+          PrefsService.set(name, store.metricsPrefsBackup[name]);
+        });
+      }
     }
   },
 
   destroy: function() {
-    // Stop the ping timer, if any
-    if (pingTimer) {
-      clearTimeout(pingTimer);
-    }
     Events.off(EVENT_SEND_METRIC, Metrics.onExperimentPing);
+    Events.off(EVENT_RECEIVE_VARIANT_DEFS, Metrics.onReceiveVariantDefs);
   },
 
-  maybePingTelemetry: function() {
-    const now = Date.now();
-    const shouldPing =
-      // Fresh install, no timestamp. Ping immediately.
-      (!store.lastTelemetryPingTimestamp) ||
-      // Subsequent pings should go out after PING_INTERVAL
-      (now - store.lastTelemetryPingTimestamp > PING_INTERVAL);
-
-    if (shouldPing) {
-      store.lastTelemetryPingTimestamp = now;
-      Metrics.pingTelemetry();
-    }
-
-    pingTimer = setTimeout(
-      Metrics.maybePingTelemetry,
-      PING_CHECK_INTERVAL
-    );
-  },
-
-  pingTelemetry: function() {
+  pingTelemetry: function(object, eventName, eventTimestamp) {
+    const payload = {
+      timestamp: (Date.now() - store.browserLoadedTimestamp) / 1000,
+      test: self.id,
```

Comment:
I'm missing it in my docs, but similar to the experiment ping, this is supposed to have a `version` value too.  Please can you add:

    `version: self.version`

sorry for missing that in the docs